simulation/traffic_simulator/test/src/data_type/test_lanelet_pose.cpp

```diff
@@ -0,0 +1,384 @@
+ */
+TEST_F(CanonicalizedLaneletPoseTest, copyAssignment)
+{
+  const CanonicalizedLaneletPose pose(
+    traffic_simulator::helper::constructLaneletPose(120659, 0.0, 0.0), hdmap_utils);
+  CanonicalizedLaneletPose pose_assign(
+    traffic_simulator::helper::constructLaneletPose(34468, 0.0, 0.0), hdmap_utils);
+
+  pose_assign = pose;
+
+  EXPECT_LANELET_POSE_EQ(
+    static_cast<traffic_simulator::LaneletPose>(pose),
+    static_cast<traffic_simulator::LaneletPose>(pose_assign));
+}
+
+/**
+ * @note Test conversion operator behavior using static_cast<LaneletPose>
+ */
+TEST_F(CanonicalizedLaneletPoseTest, conversionLaneletPose)
+{
+  const CanonicalizedLaneletPose pose(
+    traffic_simulator::helper::constructLaneletPose(120659, 0.0, 0.0), hdmap_utils);
+  const traffic_simulator::LaneletPose pose_casted =
+    static_cast<traffic_simulator::LaneletPose>(pose);
+
+  EXPECT_EQ(pose_casted.lanelet_id, 120659);
+  EXPECT_DOUBLE_EQ(pose_casted.s, 0.0);
+  EXPECT_DOUBLE_EQ(pose_casted.offset, 0.0);
+  EXPECT_VECTOR3_EQ(pose_casted.rpy, geometry_msgs::msg::Vector3());
+}
+
+/**
+ * @note Test conversion operator behavior using static_cast<Pose>
+ */
+TEST_F(CanonicalizedLaneletPoseTest, conversionPose)
+{
+  const CanonicalizedLaneletPose pose(
+    traffic_simulator::helper::constructLaneletPose(120659, 0.0, 0.0), hdmap_utils);
+
+  const geometry_msgs::msg::Pose pose1 =
+    makePose(makePoint(3822.3815, 73784.9618, -1.761), makeQuaternionFromYaw(2.060578777273));
+
+  EXPECT_POSE_NEAR(static_cast<geometry_msgs::msg::Pose>(pose), pose1, 0.01);
+}
+
+/**
+ * @note Test function behavior when alternative poses are present
+ */
+TEST_F(CanonicalizedLaneletPoseTest, hasAlternativeLaneletPose_true)
+{
+  const CanonicalizedLaneletPose pose(
+    traffic_simulator::helper::constructLaneletPose(120659, -10.0, 0.0), hdmap_utils);
+
+  EXPECT_TRUE(pose.hasAlternativeLaneletPose());
+}
+
+/**
+ * @note Test function behavior when alternative poses are absent
+ */
+TEST_F(CanonicalizedLaneletPoseTest, hasAlternativeLaneletPose_false)
+{
+  const CanonicalizedLaneletPose pose(
+    traffic_simulator::helper::constructLaneletPose(120659, 10.0, 0.0), hdmap_utils);
+
+  EXPECT_FALSE(pose.hasAlternativeLaneletPose());
+}
+
+/**
+ * @note Test function behavior when there are no lanelet_poses
+ */
+TEST_F(CanonicalizedLaneletPoseTest, getAlternativeLaneletPoseBaseOnShortestRouteFrom_empty)
+{
+  const CanonicalizedLaneletPose pose(
+    traffic_simulator::helper::constructLaneletPose(120659, 20.0, 0.0), hdmap_utils);
+  const auto from1 = traffic_simulator::helper::constructLaneletPose(34603, 10.0, 0.0);
+  const auto from2 = traffic_simulator::helper::constructLaneletPose(34579, 10.0, 0.0);
+
+  const auto result1 = pose.getAlternativeLaneletPoseBaseOnShortestRouteFrom(from1, hdmap_utils);
+  const auto result2 = pose.getAlternativeLaneletPoseBaseOnShortestRouteFrom(from2, hdmap_utils);
+
+  ASSERT_TRUE(result1);
+  ASSERT_TRUE(result2);
```

Comment:
use has_value() to avoid implicit conversion from std::optional to bool

Reply:
done